src/6model/reprs/MVMStaticFrame.c

```diff
@@ -361,7 +361,8 @@ static void describe_refs(MVMThreadContext *tc, MVMHeapSnapshotState *ss, MVMSTa
             if (body->spesh_candidates[i].sg) {
                 MVMCollectable **c_ptr;
                 MVM_spesh_graph_mark(tc, body->spesh_candidates[i].sg, ss->gcwl);
-                while (c_ptr = MVM_gc_worklist_get(tc, ss->gcwl)) {
+                c_ptr = MVM_gc_worklist_get(tc, ss->gcwl);
+                while (c_ptr) {
```

Comment:
those completly change the meaning of those `while` (executed once vs executed at each iteration).

I think the correct way to fix the assignment warning is to add a pair of parentheses `while ((a = 1))` for GCC & clang.